### ISSUE TITLE
Everywhere in system date is correct, but waybar shows UTC time

### ISSUE THREAD

**Issue by Zilibobka-S** [Alexays/Waybar#3575]
[PARTIALLY SOLVED!]
Waybar clock: ⟩ waybar -l trace
[2024-09-07 19:16:59.856] [debug] Try expanding: $XDG_CONFIG_HOME/waybar/config
[2024-09-07 19:16:59.856] [debug] Try expanding: $XDG_CONFIG_HOME/waybar/config.jsonc
[2024-09-07 19:16:59.856] [debug] Try expanding: $HOME/.config/waybar/config
[2024-09-07 19:16:59.856] [debug] Try expanding: $HOME/.config/waybar/config.jsonc
[2024-09-07 19:16:59.856] [debug] Found config file: $HOME/.config/waybar/config.jsonc
[2024-09-07 19:16:59.856] [info] Using configuration file /home/zilibobka/.config/waybar/config.jsonc
[2024-09-07 19:16:59.858] [info] Discovered appearance 'dark'
[2024-09-07 19:16:59.858] [debug] Try expanding: $XDG_CONFIG_HOME/waybar/style-dark.css
[2024-09-07 19:16:59.858] [debug] Try expanding: $XDG_CONFIG_HOME/waybar/style.css
[2024-09-07 19:16:59.858] [debug] Try expanding: $HOME/.config/waybar/style-dark.css
[2024-09-07 19:16:59.858] [debug] Try expanding: $HOME/.config/waybar/style.css
[2024-09-07 19:16:59.858] [debug] Found config file: $HOME/.config/waybar/style.css
[2024-09-07 19:16:59.858] [info] Using CSS file /home/zilibobka/.config/waybar/style.css
[2024-09-07 19:16:59.863] [debug] Output detection done: HDMI-A-1 (Philips Consumer Electronics Company PHL 222V8 ZV02102004927)
[2024-09-07 19:16:59.864] [warning] Mapping is not an object
[2024-09-07 19:16:59.867] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:16:59.881] [debug] GTK widget tree:
window#waybar.background.bottom.HDMI-A-1..mode-default:dir(ltr)
  decoration:dir(ltr)
  box.horizontal:dir(ltr)
    box.horizontal.modules-left:dir(ltr)
      widget:dir(ltr)
        box#workspaces.horizontal.module:dir(ltr)
      widget:dir(ltr)
        label#custom-right-arrow-dark.module:dir(ltr)
    box.horizontal.modules-center:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#clock.1.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-light.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#clock.2.module:dir(ltr)
      widget:dir(ltr)
        label#custom-right-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#custom-right-arrow-light.module:dir(ltr)
      widget:dir(ltr)
        label#clock.3.module:dir(ltr)
      widget:dir(ltr)
        label#custom-right-arrow-dark.module:dir(ltr)
    box.horizontal.modules-right:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#pulseaudio.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-light.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#memory.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-light.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#cpu.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-light.module:dir(ltr)
      widget:dir(ltr)
        label#custom-left-arrow-dark.module:dir(ltr)
      widget:dir(ltr)
        label#disk.module:dir(ltr)

[2024-09-07 19:16:59.882] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:16:59.882] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:16:59.882] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:16:59.985] [info] Bar configured (width: 1920, height: 26) for output: HDMI-A-1
[2024-09-07 19:17:00.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:00.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:00.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:01.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:02.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:03.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:04.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
[2024-09-07 19:17:05.000] [warning] Timezone: Europe/Moscow. std::chrono::tzdb: cannot locate zone: Europe/Moscow
^C[2024-09-07 19:17:05.722] [info] Quitting.

Timedatectl output:                Local time: Sat 2024-09-07 19:18:02 MSK
                 Universal time: Sat 2024-09-07 16:18:02 UTC
                 RTC time: Sat 2024-09-07 16:18:02
                Time zone: Europe/Moscow (MSK, +0300)
                 System clock synchronized: yes
                 NTP service: active
                 RTC in local TZ: no

 Waybar config:              
        
[config.jsonc](https://github.com/user-attachments/files/16919096/config.jsonc)

you can downgrade tzdata package, it helps

**Comment by ignamartinoli** [comment]
Exactly same problem with America/Argentina/Cordoba

**Comment by barksjn** [comment]
Same: America/Indiana/Indianapolis

If it helps I use arch testing repositories and tzdata has an update from the 6th. If I revert the update the waybar clock works right.

**Comment by pkreuzt** [comment]
See:

https://github.com/Alexays/Waybar/wiki/Module:-Clock#troubleshooting

> If using clock instead of simpleclock, the locale will default to C regardless of the locale settings. To override this behavior you have to perpend L to the formatting string. For example, {:%a %m %d} becomes {:L%a %m %d}.

**Comment by Zilibobka-S** [comment]
I downgraded tzdata package, it helps, but its still weird

**Comment by Maulve** [comment]
Can confirm with Europe/Berlin.
Downgrading tzdata fixed it though.

**Comment by Syyhis** [comment]
While using tzdata 2024b-1, std::chrono time zone database seems to be incorrect. Iterating over std::chrono::tzdb_list and printing out all the zone names per std::chrono::tzdb gives only Etc/GMT and Etc/UTC. Tested this out with a small program.
Downgrading to tzdata 2024a-2 indeed seems to fix this issue.

**Comment by TheKoTech** [comment]
> {:L%a %m %d}

Putting "L" in front doesn't work

**Comment by akgupta1337** [comment]
Having same issue!

**Comment by YuvanMichael** [comment]
me too!

**Comment by YuvanMichael** [comment]
I downgraded tzdata and it works fine now

**Comment by DistantStatic** [comment]
I can also say this is some something to do with the tzdata package.


If using Arch and Pacman:
Do note the online archive is not the only location for a rollback version.

Check /var/cache/pacman/pkg/ for the old tzdata and install using pacman -U
https://wiki.archlinux.org/title/Downgrading_packages

**Comment by lfos** [comment]
> While using tzdata 2024b-1, std::chrono time zone database seems to be incorrect. Iterating over std::chrono::tzdb_list and printing out all the zone names per std::chrono::tzdb gives only Etc/GMT and Etc/UTC. Tested this out with a small program. Downgrading to tzdata 2024a-2 indeed seems to fix this issue.

Thanks for the analysis; I think this should be reported upstream (to tzdata). There seem to be others reporting issues with tzdata 2024b-1 that may or may not be related (https://lists.iana.org/hyperkitty/list/tz@iana.org/thread/6PF5BB7342JR6PALTMQFEGU2FZ5ADW36/). @Syyhis Reporting your findings on the tzdata mailing list may increase awareness and help get out a new release faster.

Too bad this wasn't noticed and analyzed before the package left [core-testing], otherwise it'd have been easy to pull the Arch package, but I guess it's now easiest to just locally downgrade and hold the package until a new release is out.

**Comment by itshog** [comment]
Could this be a duplicate of #3024?

**Comment by lfos** [comment]
> Could this be a duplicate of #3024?

I looks like the original issue described in #3024 is different, but the thread was revived with a duplicate of this issue. So - no, this is not a duplicate, but the second half of the discussion in #3024 should instead have gone into a new bug / here.

**Comment by itshog** [comment]
> > Could this be a duplicate of #3024?
> 
> I looks like the original issue described in #3024 is different, but that thread was revived recently discussing the same issue as reported here. So - no, this is not a duplicate, but the second half of the discussion in #3024 should instead have gone into a new bug / here.

Thanks for the clarification, I believed this was the was the case but I wasn't sure about it

**Comment by NuclearC** [comment]
Downgrade to tzdata 2024-a2 fixes the problem.

**Comment by UntitledHam** [comment]
I am having this same issue

**Comment by chardoncs** [comment]
+1

**Comment by AMIRIOX** [comment]
same issue    
```bash
$ downgrade tzdata # choose 2024a-2
```
restart waybar  
done

**Comment by christiansacks** [comment]
Thanks for the info about downgrading tzdata 👍 I can confirm that worked for me in Arch also;

```
$ ls -lhrt /var/cache/pacman/pkg/tzdata-*
-rw-r--r-- 1 root root  310 Feb  2  2024 /var/cache/pacman/pkg/tzdata-2024a-1-x86_64.pkg.tar.zst.sig
-rw-r--r-- 1 root root 387K Feb  2  2024 /var/cache/pacman/pkg/tzdata-2024a-1-x86_64.pkg.tar.zst
-rw-r--r-- 1 root root  310 May  1 19:43 /var/cache/pacman/pkg/tzdata-2024a-2-x86_64.pkg.tar.zst.sig
-rw-r--r-- 1 root root 350K May  1 19:43 /var/cache/pacman/pkg/tzdata-2024a-2-x86_64.pkg.tar.zst
-rw-r--r-- 1 root root  310 Sep  6 15:08 /var/cache/pacman/pkg/tzdata-2024b-1-x86_64.pkg.tar.zst.sig
-rw-r--r-- 1 root root 347K Sep  6 15:08 /var/cache/pacman/pkg/tzdata-2024b-1-x86_64.pkg.tar.zst
```
I installed the 2nd to latest (save ver as mentioned above - 2024a-2
```
$  sudo pacman -U /var/cache/pacman/pkg/tzdata-2024a-2-x86_64.pkg.tar.zst
```
Then restarted waybar and clock displays the correct time again.

**Comment by dR3b** [comment]
Nice, but downgrade is not a solution.

**Comment by TheKoTech** [comment]
> Update: Okay, I just briefly looked through the source code. It looks not likely a waybar bug.

If waybar breaks, but not any other program, then it's waybar. On my system, that's the case

**Comment by christiansacks** [comment]
> Nice, but downgrade is not a solution.

Perhaps not, but it's a solid workaround until tzdata is fixed or Waybar can create a solution to work around it for us?

**Comment by rudrakshchahal** [comment]
I dont have tzdata a-2 to downgrade what should I do
![240909_13h06m48s_screenshot](https://github.com/user-attachments/assets/aa9b1447-f048-4a2e-8bb0-3f28c1010f75)

**Comment by cluosh** [comment]
> I dont have tzdata a-2 to downgrade what should I do ![240909_13h06m48s_screenshot](https://private-user-images.githubusercontent.com/131520192/365554968-aa9b1447-f048-4a2e-8bb0-3f28c1010f75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzEyOTQsIm5iZiI6MTcyNTg3MDk5NCwicGF0aCI6Ii8xMzE1MjAxOTIvMzY1NTU0OTY4LWFhOWIxNDQ3LWYwNDgtNGEyZS04YmIwLTNmMjhjMTAxMGY3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQwODM2MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZDk2MzM4ZDRlYzExMTRhZDlmNTI3ZDUwZWVkNzQwY2MwZjY3MTcyMWUxYjkyOTE1ZTNkY2QyMzMzZWQ5MDM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Tp8VmeixaoQI-83M2Tdo-9FgrGktHjFBZICvEqflBf8)

You can fetch it from the Arch archive: https://archive.archlinux.org/packages/t/tzdata/tzdata-2024a-2-x86_64.pkg.tar.zst

**Comment by thyeun** [comment]
Why it only happen on waybar (clock module)? swaylock working fine too.

**Comment by steveWang** [comment]
> While using tzdata 2024b-1, std::chrono time zone database seems to be incorrect. Iterating over std::chrono::tzdb_list and printing out all the zone names per std::chrono::tzdb gives only Etc/GMT and Etc/UTC. Tested this out with a small program. Downgrading to tzdata 2024a-2 indeed seems to fix this issue.

> +1
> 
> Update: Okay, I just briefly looked through the source code. It looks not likely a waybar bug. Waybar just gets the datetime through std library. That's probably why rolling back tzdata fixes the issue. (idk I might be wrong...)

Signal-boosting https://github.com/Alexays/Waybar/issues/3570#issuecomment-2336837532 -- this is a result of some new syntax in the tzdata.zi file that the main C++ standard library implementations don't yet support (fixed but not released in libstdc++ that comes with gcc, and [not yet fixed in libc++](https://github.com/llvm/llvm-project/issues/107779) that comes with llvm / clang).

There's now an bug for this on the tzdata issue tracker, where I've mentioned that other packages might be affected.

> Why it only happen on waybar (clock module)? swaylock working fine too.

Swaylock is written in C, and does not use the C++ standard library.

**Comment by thyeun** [comment]
@steveWang No wonder ```hyrlock``` affected too, since it is based on ```cpp``` library.

So any news from tzdata will fix ASAP?

**Comment by littleblack111** [comment]
the `date` command doesnt seem affected

**Comment by SolonRubens** [comment]
Having the same issue. Downgrading tzdata fixed it. For everyone who also never downgraded a package before (Arch) I recommend using the downgrade script from the AUR which makes downgrading pretty easy

**Comment by littleblack111** [comment]
@SolonRubens which version you downgraded it to? 2024a?

**Comment by alekgr** [comment]
I think i'll just wait till waybar or tzdata fixes it . @Alexays  any ideas

**Comment by moralrebuild** [comment]
please fix ASAP.

**Comment by psi4j** [comment]
> @SolonRubens which version you downgraded it to? 2024a?

Use the downgrade command and select this one:
![image](https://github.com/user-attachments/assets/35fc9356-a86f-4070-8f1f-9934aace6a11)

**Comment by Matra-Master** [comment]
Note for anywone with issues on this. Instead of downgrading a package you can also make a custom plugin that replaces clock functions. Depending on your setup it may be harder, but as an option for me that only look at the time and don't use it's hover calendar it was more than enough.
```
    "custom/clock": {
        "format": "  {}",
        "interval": 60,
        "exec": "date +'%d %a %H:%M'",
    },
```
Hope this helps while we wait

**Comment by hbacelar8** [comment]
> > @SolonRubens which version you downgraded it to? 2024a?
> 
> Use the downgrade command and select this one: ![image](https://private-user-images.githubusercontent.com/116757358/365759965-35fc9356-a86f-4070-8f1f-9934aace6a11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5MDk5NTcsIm5iZiI6MTcyNTkwOTY1NywicGF0aCI6Ii8xMTY3NTczNTgvMzY1NzU5OTY1LTM1ZmM5MzU2LWE4NmYtNDA3MC04ZjFmLTk5MzRhYWNlNmExMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxOTIwNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMzgzODhiMTM1NGY1MGM2N2M1YWIxNTI4NmEzY2RkZjU0NGY1NmIxNGU4NzEzOTQ3ODI4ZWQ4MDU3ZWE3OTY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QWV__iwRH1tCznn7AFaSYPZDc1Hs111fLr5iTVJCJOc)

which downgrade command is this?

**Comment by ClaimedLizard** [comment]
> > > @SolonRubens which version you downgraded it to? 2024a?
> > 
> > 
> > Use the downgrade command and select this one: ![image](https://private-user-images.githubusercontent.com/116757358/365759965-35fc9356-a86f-4070-8f1f-9934aace6a11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5MDk5NTcsIm5iZiI6MTcyNTkwOTY1NywicGF0aCI6Ii8xMTY3NTczNTgvMzY1NzU5OTY1LTM1ZmM5MzU2LWE4NmYtNDA3MC04ZjFmLTk5MzRhYWNlNmExMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxOTIwNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMzgzODhiMTM1NGY1MGM2N2M1YWIxNTI4NmEzY2RkZjU0NGY1NmIxNGU4NzEzOTQ3ODI4ZWQ4MDU3ZWE3OTY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QWV__iwRH1tCznn7AFaSYPZDc1Hs111fLr5iTVJCJOc)
> 
> which downgrade command is this?

It's an AUR package. Here's the link: https://aur.archlinux.org/packages/downgrade

**Comment by oscarpax** [comment]
> > > @SolonRubens which version you downgraded it to? 2024a?
> which downgrade command is this?

`sudo pacman -U /var/cache/pacman/pkg/tzdata-2024a-2-x86_64.pkg.tar.zst`

On Arch, the above command downgrades `tzdata` package to previous version, given that you had not cleared your pacman cache after the version was released.

[https://wiki.archlinux.org/title/Pacman](url)

**Comment by psi4j** [comment]
> > > @SolonRubens which version you downgraded it to? 2024a?
> > 
> > 
> > Use the downgrade command and select this one: ![image](https://private-user-images.githubusercontent.com/116757358/365759965-35fc9356-a86f-4070-8f1f-9934aace6a11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5MDk5NTcsIm5iZiI6MTcyNTkwOTY1NywicGF0aCI6Ii8xMTY3NTczNTgvMzY1NzU5OTY1LTM1ZmM5MzU2LWE4NmYtNDA3MC04ZjFmLTk5MzRhYWNlNmExMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxOTIwNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMzgzODhiMTM1NGY1MGM2N2M1YWIxNTI4NmEzY2RkZjU0NGY1NmIxNGU4NzEzOTQ3ODI4ZWQ4MDU3ZWE3OTY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QWV__iwRH1tCznn7AFaSYPZDc1Hs111fLr5iTVJCJOc)
> 
> which downgrade command is this?

![image](https://github.com/user-attachments/assets/4a498110-7cba-460a-9efe-1c0007c9e2f7)

**Comment by brighamdent** [comment]
Same thing happened to me I just downgraded I tzdata and things work again.

**Comment by JosTheDude** [comment]
Issue is apparent with `America/New_York`

Here are steps I took to fix:

`yay -Syu downgrade` # or preferred AUR helper
`sudo downgrade tzdata`

Pick 2024-a-2 and restart waybar

**Comment by frod0r** [comment]
This might be obvious, but for the time being a workaround I prefer over downgrading packages, is to just create a small custom widget

```json
    "custom/clock": {
      "exec": "date +%H:%M:%S",
      "interval": 1
    }
```

**Comment by jwakely** [comment]
> Thanks for the analysis; I think this should be reported upstream (to tzdata). There seem to be others reporting issues with tzdata 2024b-1 that may or may not be related (https://lists.iana.org/hyperkitty/list/tz@iana.org/thread/6PF5BB7342JR6PALTMQFEGU2FZ5ADW36/). @Syyhis Reporting your findings on the tzdata mailing list may increase awareness and help get out a new release faster.

This is not an upstream tzdata problem, and is not related to the "Apr" vs "April" problems already reported upstream.

This problem is caused by a change in how Arch Linux packages tzdata, which seems to have changed between the 2024a-2 and 2024b-1 packages. The new package contains %z strings which GCC trunk handles correctly, but gcc-14 does not (I need to backport a fix to gcc-14).

**Comment by chardoncs** [comment]
> > Thanks for the analysis; I think this should be reported upstream (to tzdata). There seem to be others reporting issues with tzdata 2024b-1 that may or may not be related (https://lists.iana.org/hyperkitty/list/tz@iana.org/thread/6PF5BB7342JR6PALTMQFEGU2FZ5ADW36/). @Syyhis Reporting your findings on the tzdata mailing list may increase awareness and help get out a new release faster.
> 
> This is not an upstream tzdata problem, and is not related to the "Apr" vs "April" problems already reported upstream.
> 
> This problem is caused by a change in how Arch Linux packages tzdata, which seems to have changed between the 2024a-2 and 2024b-1 packages. The new package contains %z strings which GCC trunk handles correctly, but gcc-14 does not (I need to backport a fix to gcc-14).

https://gitlab.archlinux.org/archlinux/packaging/packages/tzdata/-/issues/1

That makes sense. Also, there has been someone reported this issue on Arch's GitLab.

**Comment by thyeun** [comment]
So who should fix it? Look like the baby was push here and there, and nobody want to take the baby. 😅😅😅

**Comment by chardoncs** [comment]
> So who should fix it? Look like the baby was push here and there, and nobody want to take the baby. 😅😅😅

It is definitely not Waybar's issue. Waybar's source code is fine.

Also @jwakely just said the issue is not from the tzdata upstream. It should be the Arch package if this issue only affects Arch users.

**Comment by jwakely** [comment]
I need to backport the GCC fix to the release branches, but that will take some time to reach users.

**Comment by hbacelar8** [comment]
> I need to backport the GCC fix to the release branches, but that will take some time to reach users.
> 
> Edit: the backports are done

so can we update tzdata now without any issues?

**Comment by jwakely** [comment]
No, not until you're running a new build of GCC. I haven't updated the Arch gcc package or installed it on your system.

**Comment by Kylar514** [comment]
Its fixed now with the latest update on gcc.  updating arch and restarting waybar will fix the issue now.... I just updated to gcc 14.2

**Comment by ThorpeJosh** [comment]
Can confirm, updating to tzdata `2024b-2` from `2024b-1` has fixed the issue for me.

**Comment by Dvd-Znf** [comment]
I think this issue may be safely closed

**Comment by salfel** [comment]
issue should be fixed on most package managers, at least for arch it was the case

**Comment by jwakely** [comment]
> issue should be fixed on most package managers, at least for arch it was the case

Have any other distros packaged gcc-14 or gcc-13 snapshots from **yesterday** yet? I don't see any except Arch. Fedora and Ubuntu both updated their gcc-14.2.1 a few days ago, and so don't have the fix. Similarly for Gentoo.

**Comment by jwakely** [comment]
> Seems like it has been fixed in tzdata-2024b-2!

The ArchLinux tzdata-2024b-2 package only [changes](https://gitlab.archlinux.org/archlinux/packaging/packages/tzdata/-/commit/259ad2a45b5121a9c94f53e1a380790af9a8d046) "April" to "Apr", which fixes a problem with some other tzdata consumers, but doesn't make any difference for this issue. This issue is fixed by a change in GCC.

**Comment by commandline-be** [comment]
this is very annyoing and looks trivial enough
 [warning] Timezone: CET. std::chrono::tzdb: cannot locate zone <TZHERE>
 
 experiencing same issue on SuSe

**Comment by jwakely** [comment]
> this is very annyoing and looks trivial enough [warning] Timezone: CET. std::chrono::tzdb: cannot locate zone
> 
> experiencing same issue on SuSe

SuSe needs to update its GCC package

**Comment by eoli3n** [comment]
I don't get why so much users are happy with a fix that requires to downgrade a package, isn't it a workaround ?

**Comment by jwakely** [comment]
The fix is to update GCC, not downgrade anything.

If your distro doesn't have a new GCC that can work with the new tzdata, then you might need to use old versions of the two that are compatible with each other. But that's your distro's problem, for not giving you up to date packages that work well together.

**Comment by thyeun** [comment]
Who still solve this by downgrade waybar version? This is nothing related with waybar, it is because of the GCC compatible issue, just get the latest version of GCC, it will solve it automatically.

**Comment by psi4j** [comment]
> I don't get why so much users are happy with a fix that requires to downgrade a package, isn't that a workaround ?
> 
> On void linux:
> ```
> [*] tzdata-2025a_1
> [*] gcc-13.2.0_2
> [*] Waybar-0.11.0_2
> ```
> 
> Timezone seems ignored

Maybe let the void team know about the issue. I don't know how they do things over there so I can't tell you if they'd be able to fix it anytime soon, but I can tell you that on Arch, this is no longer an issue.

**Comment by eoli3n** [comment]
I face this now : https://github.com/Alexays/Waybar/issues/3891

**Comment by eoli3n** [comment]
https://github.com/void-linux/void-packages/issues/54018

**Comment by lady-melinoe** [comment]
Currently on void-linux with gcc14, rebuilding Waybar on gcc14 didn't fix it for me. Currently using a custom widget running `date '+%I:%M %p'` as my clock.

**Comment by vimscientist69** [comment]
For anyone on ARM processor (aarch64), you can downgrade using the following command:

Download old version of tzdata: https://archive.archlinux.org/packages/t/tzdata/tzdata-2024a-2-x86_64.pkg.tar.zst
Run:

```bash
sudo pacman -U ~/Downloads/tzdata-2024a-2-x86_64.pkg.tar.zst --arch x86_64
```
